src/uint.rs

```diff
@@ -910,34 +910,35 @@ macro_rules! construct_uint {
 			}
 		}
 
-		#[cfg(feature="std")]
 		impl ::core::fmt::Debug for $name {
 			fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
 				::core::fmt::Display::fmt(self, f)
 			}
 		}
 
-		#[cfg(feature="std")]
 		impl ::core::fmt::Display for $name {
 			fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
 				if self.is_zero() {
 					return write!(f, "0");
 				}
 
-				let mut s = String::new();
+				let mut buf = [0_u8, $n_words*8];
+				let mut i = buf.len() - 1;
 				let mut current = *self;
 				let ten = $name::from(10);
 
 				while !current.is_zero() {
-					s = format!("{}{}", (current % ten).low_u32(), s);
+					let digit = (current % ten).low_u32() as u8;
+					buf[i] = digit + b'0';
 					current = current / ten;
+					i -= 1;
 				}
 
-				write!(f, "{}", s)
+				let s = unsafe {::core::str::from_utf8_unchecked(&buf[i..])};
```

Comment:
unjustified `unsafe`

Reply:
String of numbers is guaranteed to be a valid UTF8 string.